### ISSUE TITLE
Computing resources required to train the model

### ISSUE THREAD

**Issue by coderjingluo** [erikl2/nested-learning#1]
When I train models of 340M or 760M size using an RTX 4090D (24GB), the Python script below gets killed. Is this due to insufficient computational resources?
python experiments/train_lm.py \
    --model hope \
    --size 760M \
    --dataset wikitext \
    --optimizer deep_momentum

**Comment by coderjingluo** [comment]
root@autodl-container-93b9118200-ed3a7169:~/ai/nested-learning# python experiments/train_lm.py --model hope --size 340M --dataset wikitext --dataset_config wikitext-103-raw-vl --optimizer deep_momentum --batch_size 8 --max_length 1024 --epochs 10 --device cuda 
Training HOPE model: 340M
  dim=512, n_layers=12, n_heads=8
Number of parameters: 115.0M
Killed

GPU：32GB

**Comment by YaoxinHuang** [comment]
> `python experiments/benchmark_lambada.py --test`运行时默认的340M模型在推理时占用12G显存，但是我跑`python experiments/benchmark_lambada.py`时候一张A100-80GB的卡都报OOM，不明觉厉。另外代码中有不少和论文中不一致的地方。

`python experiments/benchmark_lambada.py`会报OOM完全是因为作者了一个`当前句=torch.cat(前一句，后一句)`的递归逻辑，显存一路飙，鉴于目前版本作者还没有实现“分布式训练”，难以想象是什么GPU支持如此大的内存占用。
另外，使用`python experiments/train_lm.py`训练340M的模型大约占用26GB左右的显存容量，仅供参考。事实上在运行这份代码时候你会遇见另一个问题：optimizer没有to(device)会报错。需要把optimizer中的mlp也放到cuda上去。

**Comment by TomLucidor** [comment]
Just test this against Modified-NanoGPT
